### DISCUSSION TITLE
How to use useMenu? It doesn't produce the desired results

### DISCUSSION THREAD

**Discussion by benjick** [algolia/instantsearch#5338] [Q&A] [answered]
If I look at the og MenuSelect at https://instantsearchjs.netlify.app/stories/?path=/story/refinements-menuselect--default I can see how I want it to work. You can select A and then select B. But reading the docs for the react-hook library https://www.algolia.com/doc/api-reference/widgets/menu-select/react-hooks/ I can't get it to produce the same result.

Using the example, as soon as I run `refine` on an item; `items` will just contain that one item. How can I produce the same result as for `MenuSelect`?

Full code example:
```ts
function MenuSelect({
  label,
  ...props
}: UseMenuProps & {
  label: string;
}) {
  const { items, refine } = useMenu(props);
  const [selected, setSelected] = useState("");

  useEffect(() => {
    if (selected) {
      refine(selected);
    }
  }, [selected]);

  return (
    <select value={selected} onChange={(e) => setSelected(e.target.value)}>
      <option value="">Välj {label}</option>
      {items.map((i) => (
        <option key={i.value} value={i.value}>
          {i.label} ({i.count})
        </option>
      ))}
    </select>
  );
}
```

**Comment by Haroenv** [comment, marked as answer]
The code in the documentation is indeed wrong, I'll fix it (edit: this code didn't come from the doc apparently). This works: 

```js
function MenuSelect({
  label,
  ...props
}: UseMenuProps & {
  label: string;
}) {
  const { items, refine } = useMenu(props);

  const currentRefinement = (
    items.find((item) => item.isRefined) || { value: '' }
  ).value;

  return (
    <select value={currentRefinement} onChange={(e) => refine(e.target.value)}>
      <option value="">Välj {label}</option>
      {items.map((i) => (
        <option key={i.value} value={i.value}>
          {i.label} ({i.count})
        </option>
      ))}
    </select>
  );
}
```

essentially it avoids refining before a value is available

demo: https://codesandbox.io/s/optimistic-swartz-pt7els?file=/src/App.tsx:1924-2443

**Reply by benjick** [reply]
Hey!

Thanks for the reply. It might not a problem with the code then, because I can see it working in your example but it still doesn't work in my code.

Before refining:
<img width="247" alt="Screenshot 2022-10-24 at 11 24 08" src="https://user-images.githubusercontent.com/430872/197494086-1263c4ef-6399-4c34-8e67-0dbe0a9f28f1.png">

After refining:
<img width="175" alt="Screenshot 2022-10-24 at 11 24 13" src="https://user-images.githubusercontent.com/430872/197494096-7f2ed291-1615-4c39-acb1-2415ca62c8ab.png">

Could it be because I'm using Meilisearch?

**Reply by Haroenv** [reply]
That indeed would be Meili not returning the facets disjunctively. Sorry!
